packages/api/cms-api/src/content-generation/azure-open-ai-content-generation.service.ts

```diff
@@ -1,19 +1,56 @@
 import { AzureKeyCredential, ChatRequestMessage, OpenAIClient } from "@azure/openai";
-import { Injectable } from "@nestjs/common";
+import { Inject, Injectable } from "@nestjs/common";
 
 import { FilesService } from "../dam/files/files.service";
+import { AZURE_OPEN_AI_CONTENT_GENERATION_SERVICE_CONFIG } from "./content-generation.constants";
+import { ContentGenerationServiceInterface } from "./content-generation-service.interface";
 
-export type AzureOpenAiContentGenerationConfig = {
+export type AzureOpenAiContentGenerationServiceConfig = AzureOpenAiConfig | ConfigByMethod;
```

Comment:
is it really a use-case that you need a different deploymentId/apiKey/apiUrl per method?

Reply:
In this specific case: Probably not. Title and Alt Text will usually use the same model.

Generally: Yes. 

- The deploymentId specifies the model in use. Different tasks require different models (e.g. GPT-4o for complicated tasks, GPT-3.5 for simple tasks to save time and money, Dall-E for image processing). 

- Not all models are available in all Azure regions. E.g. GPT-4o is available in swedencentral but not francecentral (as you can see [here](https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models#gpt-4-and-gpt-4-turbo-model-availability). Different region -> different URL (and API key)

Since we don't really know yet, which features we want to build with this, I'm trying to "future-proof" the config. I'm aware that "premature optimization is the root of all evil". But if we later realize that the config isn't flexible enough, we need another major version (and probably some ugly non-breaking solution in this version) to fix it.